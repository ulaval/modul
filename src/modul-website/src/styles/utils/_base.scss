html,
body,
.modul-body {
    background: $modul-color--body;
    height: 100%;
}
