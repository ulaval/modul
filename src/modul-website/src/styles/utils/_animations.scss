.modul-u--light-template-transition {
    &-enter-active,
    &-leave-active {
        @media (min-width: #{$m-mq--min--s}) {
            transition: transform $m-transition-duration--l ease, opacity $m-transition-duration--l ease;
        }
    }

    &-enter,
    &-leave-to {
        @media (min-width: #{$m-mq--min--s}) {
            transform: translate3d(0, 44px, 0);
            opacity: 0.6;
        }
    }
}
