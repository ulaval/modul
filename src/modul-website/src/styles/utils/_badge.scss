.modul-logo__beta {
    display: inline-flex;
    align-items: center;
    height: 12px; // Magic number
    margin-left: $m-margin--s;
    text-transform: uppercase;
    font-size: 8px; // Mafic number
    font-weight: $m-font-weight--black;
    color: $modul-color--black-grey-dark;
    background: $m-color--ul-yellow;
    padding: 2px 3px;
}

.modul-logo__new {
    display: inline-flex;
    align-items: center;
    height: 12px; // Magic number
    margin-left: $m-margin--s;
    text-transform: uppercase;
    font-size: 8px; // Mafic number
    font-weight: $m-font-weight--black;
    color: $modul-color--black-grey-dark;
    background: $m-color--hint;
    padding: 2px 3px;
}
