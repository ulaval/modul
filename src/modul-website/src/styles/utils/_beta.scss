.modul-logo__beta {
    display: inline-flex;
    align-items: center;
    height: 12px; // Magic number
    margin-left: $m-margin--s;
    text-transform: uppercase;
    font-size: 8px; // Mafic number
    font-weight: $m-font-weight--black;
    color: $modul-color--black-grey-dark;
    background: $m-color--ul-yellow;
    padding: 2px 3px;
}
