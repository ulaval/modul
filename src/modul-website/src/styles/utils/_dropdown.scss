.modul-u--dropdown-light.modul-u--dropdown-light,
.modul-u--dropdown-dark.modul-u--dropdown-dark {
    .m-dropdown__input.m-dropdown__input,
    .m-input-style__body.m-input-style__body {
        padding-right: $m-padding--s;

        @include modul-page-title();
    }

    .m-input-style {
        &::before {
            border-width: 2px;

            @media (min-width: $m-mq--min--s) {
                border-width: 3px;
            }
        }

        &::after {
            @media (min-width: $m-mq--min--s) {
                height: 3px;
            }
        }
    }
}

.modul-u--dropdown-light.modul-u--dropdown-light {
    .m-input-style {
        &__body {
            input {
                color: $modul-color--black-grey-dark;
            }
        }

        &::before {
            border-color: $modul-color--black-grey-dark !important;
        }
    }

    &:not(.m--is-open) {
        .m-dropdown__arrow {
            .m-icon-button.m-icon-button {
                color: $modul-color--black-grey-dark !important;
            }
        }
    }
}

.modul-u--dropdown-dark.modul-u--dropdown-dark {
    .m-input-style {
        &__body {
            input {
                color: #fff;
            }
        }

        &::before {
            border-color: $m-color--ul-yellow !important;
        }
    }

    &:not(.m--is-open) {
        .m-dropdown__arrow {
            .m-icon-button.m-icon-button {
                color: $m-color--ul-yellow !important;
            }
        }
    }
}
