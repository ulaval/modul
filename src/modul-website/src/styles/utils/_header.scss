// ======================================================================
//  header (list, table)
// ======================================================================
.mu-header {
    position: relative;
    font-weight: $m-font-weight-semi-bold;
    padding: $m-space;
    background-color: $m-color-grey-lighter;

    @include m-font-size('md');

    p,
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        margin-top: 0;
        margin-bottom: 0;
        font-weight: $m-font-weight-semi-bold;

        @include m-font-size('md');
    }

    &.m-u--has-icon {
        padding-right: $m-touch-size;

        > :last-child {
            width: $m-touch-size;
            position: absolute;
            top: 50%;
            transform: translateY(-50%);
            right: 0;
        }
    }

    &:not(.m-u--has-icon) {
        display: flex;
        align-items: center;
        justify-content: space-between;

        @media screen and (max-width: $m-mq-max-sm) {
            flex-direction: column;
            align-items: flex-start;
        }
    }

    + ul {
        margin-top: 0;
    }

    + .mu-table {
        th {
            padding-top: $m-space;
        }
    }
}
