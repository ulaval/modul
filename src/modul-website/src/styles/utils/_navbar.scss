.modul-u--navbar {
    position: relative;
    border-top: $m-border-width--l solid $m-color--ul-yellow;

    &::after {
        position: absolute;
        bottom: $m-margin--s; // Magic number
        left: 0;
        right: 0;
        z-index: 0;
        content: '';
        height: 1px;
        background: $m-color--grey;
    }

    .m-navbar {
        position: relative;
        z-index: 1;

        // &__wrap {
        //     background: $m-color--white;
        // }

        .m-navbar-item.m--is-selected .modul-u--navbar__item {
            color: $modul-color--black-grey-dark;
        }
    }

    .m-navbar-item {
        .modul-u--navbar__item {
            padding: 25px 0; //Magic Number
            font-weight: $m-font-weight--black;
            text-transform: uppercase;
            text-decoration: none;
            color: $modul-color--grey-dark;

            &:hover,
            &:focus {
                color: $modul-color--grey-darker;
            }
        }
    }
}
