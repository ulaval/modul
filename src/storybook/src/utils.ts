export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
