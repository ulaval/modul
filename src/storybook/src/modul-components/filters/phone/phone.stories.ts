import { select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import { PHONE_NAME } from '@ulaval/modul-components/dist/filters/filter-names';
import PhoneFilterPlugin, { PhoneFilter, PhoneNumberFormat } from '@ulaval/modul-components/dist/filters/phone/phone';
import Vue from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

Vue.use(PhoneFilterPlugin);

function getBaseVueWithLocale(template: string): any {
    return {
        template,
        props: {
            text: {
                default: text('Phone number', '+14185479029')
            },
            format: {
                default: select('Format', {
                    [PhoneNumberFormat.NATIONAL]: PhoneNumberFormat.NATIONAL,
                    [PhoneNumberFormat.INTERNATIONAL]: PhoneNumberFormat.INTERNATIONAL
                }, PhoneNumberFormat.INTERNATIONAL)
            }
        },
        methods: {
            formatedPhoneNumber(text: string, format: PhoneNumberFormat): string {
                return PhoneFilter.format(text, format);
            }
        }
    };
}

function getBaseSimpleVue(template: string): any {
    return getBaseVueWithLocale(template);
}

storiesOf(`${modulComponentsHierarchyRootSeparator}${PHONE_NAME}`, module)
    .add('Default', () => getBaseSimpleVue(`
        <div>
            <div>Phone number : <span v-html="text"></span></div>
            <div>Result : <span v-html="formatedPhoneNumber(text, format)"></span></div>
        </div>`)
    );
