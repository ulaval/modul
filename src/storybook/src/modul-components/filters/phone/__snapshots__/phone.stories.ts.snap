// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|f-m-phone Default 1`] = `
<div>
  <div>
    Phone number : 
    <span>
      +14185479029
    </span>
  </div>
   
  <div>
    Result : 
    <span>
      +1 418 547 9029
    </span>
  </div>
</div>
`;
